processing/src/main/java/org/apache/druid/segment/join/HashJoinSegment.java

```diff
@@ -122,15 +122,14 @@ public void close() throws IOException
         }).orElse(true);
       }
       if (acquireFailed) {
-        CloseQuietly.close(closer);
+        CloseableUtils.closeAndWrapExceptions(closer);
         return Optional.empty();
       } else {
         return Optional.of(closer);
       }
     }
-    catch (Exception ex) {
-      CloseQuietly.close(closer);
-      return Optional.empty();
```

Comment:
This method was using `CloseQuietly` to eat exceptions, although probably not well enough since it was only eating `IOExceptions`. I think it should probably be using `closeAndSuppressExceptions`. 

Per the javadocs of `acquireReferences`:

```
...
   * IMPORTANT NOTE: to fulfill the contract of this method, implementors must return a closeable to indicate that the
   * reference can be acquired, even if there is nothing to close. Implementors should avoid allowing this method or the
   * {@link Closeable} it creates to throw exceptions.
   *
   * For callers: if this method returns non-empty, IT MUST BE CLOSED, else reference counts can potentially leak.
...
```

Reply:
Good point, changed this to suppress.